internal/otelcollector/config/metric/agent/config.go

```diff
@@ -101,12 +101,22 @@ const (
 type Processors struct {
 	config.BaseProcessors `yaml:",inline"`
 
-	DeleteServiceName           *config.ResourceProcessor `yaml:"resource/delete-service-name,omitempty"`
-	InsertInputSourceRuntime    *config.ResourceProcessor `yaml:"resource/insert-input-source-runtime,omitempty"`
-	InsertInputSourcePrometheus *config.ResourceProcessor `yaml:"resource/insert-input-source-prometheus,omitempty"`
-	InsertInputSourceIstio      *config.ResourceProcessor `yaml:"resource/insert-input-source-istio,omitempty"`
+	DeleteServiceName                     *config.ResourceProcessor `yaml:"resource/delete-service-name,omitempty"`
+	InsertInputSourceRuntime              *config.ResourceProcessor `yaml:"resource/insert-input-source-runtime,omitempty"`
+	InsertInputSourcePrometheus           *config.ResourceProcessor `yaml:"resource/insert-input-source-prometheus,omitempty"`
+	InsertInputSourceIstio                *config.ResourceProcessor `yaml:"resource/insert-input-source-istio,omitempty"`
+	DropIstioMetricsToTelemetryComponents *FilterProcessor          `yaml:"filter/drop-istio-metrics-to-internal-components,omitempty"`
```

Comment:
The naming "...metrics-to-internal components" or "...MetricsToTelemetryComponents" is a bit hard to understand. How about "drop-internal-communication"? Same for the PR description.